ods_ci/tests/Tests/0100__platform/0102__monitor_and_manage/0102__alerts/0102__alerts.robot

```diff
@@ -199,6 +200,139 @@ Verify That MT-SRE Are Not Paged For Alerts In Clusters Used For Development Or
+
+    FOR    ${metric}    IN    @{metrics_to_check}
+            Wait Until Keyword Succeeds    1m    5s
+            ...     Metric Should Be Equal To Value
+                    ...    pm_url=${RHODS_PROMETHEUS_URL}
+                    ...    pm_token=${RHODS_PROMETHEUS_TOKEN}
+                    ...    pm_query=${metric}{dspa_namespace="${PROJECT}"}
+                    ...    expected_value=1
+    END
+
+    Make Dummy GET Request To ds-pipeline-dspa Route    expected_status=404
+
+    Wait Until Keyword Succeeds    2m    5s
+    ...     Metric Should Be Equal To Value
+            ...    pm_url=${RHODS_PROMETHEUS_URL}
+            ...    pm_token=${RHODS_PROMETHEUS_TOKEN}
+            ...    pm_query=haproxy_backend_http_responses_total:burnrate5m{component="dsp", exported_namespace="${PROJECT}"}
+            ...    expected_value=0
+
+    Log To Console    "Scaling down"
+    ODS.Scale Deployment    ${PROJECT}        ds-pipeline-dspa                                     replicas=0
+    ODS.Scale Deployment    ${PROJECT}        ds-pipeline-persistenceagent-dspa                    replicas=0
+    ODS.Scale Deployment    ${PROJECT}        ds-pipeline-scheduledworkflow-dspa                   replicas=0
+
+    Log To Console    "Verifying metrics"
+    FOR    ${metric}    IN    @{metrics_to_check}
+            Wait Until Keyword Succeeds    1m    5s
+            ...     Metric Should Be Equal To Value
+                    ...    pm_url=${RHODS_PROMETHEUS_URL}
+                    ...    pm_token=${RHODS_PROMETHEUS_TOKEN}
+                    ...    pm_query=${metric}{dspa_namespace="${PROJECT}"}
+                    ...    expected_value=0
+    END
+
+    Make Dummy GET Request To ds-pipeline-dspa Route    expected_status=503
+
+    Wait Until Keyword Succeeds    1m    5s
+    ...     Metric Should Be Greater Than Value
+            ...    pm_url=${RHODS_PROMETHEUS_URL}
+            ...    pm_token=${RHODS_PROMETHEUS_TOKEN}
+            ...    pm_query=haproxy_backend_http_responses_total:burnrate5m{component="dsp", exported_namespace="${PROJECT}"}
+            ...    greater_than_value=0
+
+    Log To Console    "Verifying alerts are pending or firing"
+    @{alerts_to_check} =    Create List    Data Science Pipeline Application Unavailable
+    ...                                    Data Science Pipeline APIServer Unavailable
+    ...                                    Data Science Pipeline PersistenceAgent Unavailable
+    ...                                    Data Science Pipeline ScheduledWorkflows Unavailable
+
+    FOR    ${alert}    IN    @{alerts_to_check}
+            Prometheus.Wait Until Alert Is Pending    ${RHODS_PROMETHEUS_URL}
+            ...    ${RHODS_PROMETHEUS_TOKEN}
+            ...    RHODS Data Science Pipelines
+            ...    ${alert}
+            ...    alert-duration=120
+            ...    timeout=5 min
+    END
```

Comment:
@MarianMacik 

> The first 4 alerts have rules which look 3 minutes in the past and if there is at least one success (max value is 1), it won't start firing. So the test has to wait 3 minutes for these rules to make alerts pending and alert would start firing after additional 2 minutes, which I wanted to avoid.

this is what I meant - here we wait up to 5 minutes for each of the metrics to go into `Pending`, but it's possible that during waiting for the first one, the later ones will go from `Pending` into `Firing` and the test will fail, no?

What I'm saying is that we don't actually care whether the alert is pending or firing, they are equivalent for the purpose of this test, so it might be safer to check that the alert is in either of the states.